Add documentation to build VM for apple silicon using UTM.

eddybl opened the pull request:
Not sure if there will be too much interest, but since I put in the work I might as well document it ;)

ralphlange commented:
Looks good to me but I can't test it. @kasemir - could you give it a try?

kasemir commented:
I'm stuck at the end of the Linux install.

When it's all "complete" and I push the "Reboot" button, it will start over with the installation because the "CD" for installing is still in the virtual drive:

![Screenshot 2024-06-17 at 9 10 45 AM](https://github.com/epics-training/training-vm/assets/1932421/c9622403-2554-4a8f-b210-862405fb040b)

When I click on the "CD/DVD" and select "Clear", nothing happens.
I can press the "power" button on the running image, ignore the warning, and now the image is idle, and the virtual CD/DVD is "empty". But when I now "power on", I keep seeing this:

![Screenshot 2024-06-17 at 9 12 26 AM](https://github.com/epics-training/training-vm/assets/1932421/649d0785-519f-4360-bd27-6c8e98a72461)

Selecting the first two options very briefly shows something that I can't read, then back to the boot list.


Will start over once more after deleting this VM image, maybe I missed something.

ralphlange commented:
@eddybl Can you relate to that?

kasemir commented:
2nd attempt ended the same way, with a little more detail.

Once the installation is "complete", I cannot "Clear" the CD/DVD mount from the main UTM window. Presumably, the ISO image on the virtual CCD is "in use" and cannot be ejected.
Once I push the "Reboot" button, I can then quickly "Clear" the CD/DVD mount, but the reboot just hangs, which might be a matter of bad timing.
Once I "power off" and then reboot, there's the never ending boot up list. Sometimes I briefly see this explanation for why it won't boot:

![Screenshot 2024-06-17 at 9 39 40 AM](https://github.com/epics-training/training-vm/assets/1932421/ef7c2fcb-1c53-4d4f-81c8-2641eb3c3613)

ralphlange commented on doc/creating-vm-from-scratch.md:
```diff
@@ -5,6 +5,9 @@ for the EPICS Training from scratch.
 The recipe was set up using VirtualBox 7.0.14 on Windows 10,
 but it should work similarly on other versions and host systems.
 
+*This will not work for Silicon based Macs. Follow
```
Apple Silicon

ralphlange commented on doc/creating-vm-from-scratch.md:
```diff
@@ -5,6 +5,9 @@ for the EPICS Training from scratch.
 
+*This will not work for Silicon based Macs. Follow
+[these instructions](/doc/creating-vm-from-scratch-apple-silicon.md) for an altnernative.*
+
```
alternative

kasemir replied:
That's what I did. Install UTM, ...  as per the file in this pull request...

kasemir replied:
creating-vm-from-scratch-**apple-silicon**.md

ralphlange replied:
The text should say "Apple Silicon" instead of just "Silicon" (the file name is fine).
The word "altnernative" contains a typo.

eddybl commented:
I will have some time tomorrow in the train where I can have another look. I will try to reproduce my steps and see if it works, the one I set up during the workshop (I set it up multiple times) is still working

I will try to report back

kasemir commented:
Thank you!

Your instructions seem very clear overall, but maybe I missed on thing.
When I start UTM, there's the option to pick either "Virtualization" or "Emulation".
Since you write "choose Virtualize", I picked "Virtualization".

But then there's another button to enable Apple virtualization ("not recommended, use QEMU...").
I hadn't checked that.
Maybe you meant both
1) Pick "Virtualization"
2) Then also check that "Apple virtualization" option.

Will see how that goes...

kasemir commented:
With both UTM "Virtualization" and the "Apple virtualization" option, a "reboot" at the end of the installation gets into a "[Terminated]" state. "Power down", clear the "External Drive", start the image again, and I  get that boot loop mentioned before.
Found one more option to "Enable Rosetta on Linux", but that doesn't make a difference.

kasemir commented:
Good news: @hartmansm followed these instructions using Ubuntu 22 for ARM64 (https://mac.getutm.app/gallery) and the basic linux setup completed fine. Need to figure out if it's about Rocky vs Ubuntu, or some magic touch.

eddybl commented:
So I experimented with setting up multiple VMs over the last few days, both with QEMU and apple virtuatlization and sometimes it just does not work.

But I also got Rocky 9.4 with Apple Virtualization to work. 
So "work" or "not work" is meant in a sense that after installation Rocky linux properly boots.
If it does not work, it still boots into installation GRUB settings.

If it works, I do not have to change the boot order or VM configuration or anything.
But I do not have a good explanation for how and why this sometimes works and sometimes does not. At the moment I was doing a lot of work on my laptop parallel to a workshop, so I suspect that in my case maybe some issues happen which corrupts the installation process somehow...

kasemir commented:
Thanks for following up. In a way that's good news. We all get it to install and then reach the GRUP boot menu. Next, however, it either boots or gives some error.
As far as you can tell by now, does your "working" setup always boot, and does the "not working" setup always fail?
Or it that up to luck each time you try it?
For me, I seem stuck with "not working".

kasemir reviewed (approved):
As for this PR, I think the instructions are good.
Maybe add a comment that the process isn't 100% deterministic at this time, and then we keep updating it as we learn more

eddybl commented:
I either get a VM which never boots into the installed OS or I get a VM which always properly starts

eddybl commented:
I fixed the two typos Ralph mentioned, added the note, that it does also work on Ubuntu and added a comment that there might be an issue after installation of the OS.

I also added a comment regarding Apple Virtualisation option

ralphlange commented:
Thanks everyone!!
